Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc

```diff
@@ -458,15 +468,21 @@ AlCaIsoTracksProducer::select(edm::Handle<edm::TriggerResults>& triggerResults,
 						  trkCaloDirections,
 						  a_charIsoR_,
 						  nNearTRKs, false);
+      HcalDetId detId = (HcalDetId)(trkDetItr->detIdHCAL);
+      int       ieta = detId.ietaAbs();
+      double eIsolation = (useEtaVaryingCut_) ? (maxRestrictionP_*exp(slopeRestrictionP_*((double)(ieta)))) : 0.0;
+      if (eIsolation < eIsolate1_) eIsolation = eIsolate1_;
+      if (eIsolation < eIsolate2_) eIsolation = eIsolate2_;
 #ifdef DebugLog
```

Comment:
Thanks @bsunanda for the useful pointer to documentation and http://cms.cern.ch/iCMS/user/noteinfo?cmsnoteid=CMS%20DN-2016/029

In this piece of code, since eIsolate1_ < eIsolate2_ by construction,

eIsolation will:

   . stay as it comes out of the formula if it's > eIsolate2_
   . be set to eIsolate2_ in all other cases

I don't think this is the expected behaviour ?